sql/core/src/test/scala/org/apache/spark/sql/UserDefinedTypeSuite.scala

```diff
@@ -194,4 +293,35 @@ class UserDefinedTypeSuite extends QueryTest with SharedSQLContext with ParquetT
     // call `collect` to make sure this query can pass analysis.
     pointsRDD.as[MyLabeledPoint].map(_.copy(label = 2.0)).collect()
   }
+
+  test("SPARK-19311: UDFs disregard UDT type hierarchy") {
+    UDTRegistration.register(classOf[IExampleBaseType].getName,
```

Comment:
i tend to leave them, but remove the @SQLUserDefinedType, so we have a test that uses UDTRegistration